src/client/kinto_http.rs

```diff
@@ -101,16 +91,21 @@ pub fn get_changeset(
     server: &str,
     bid: &str,
     cid: &str,
-    expected: u64,
+    expected: Option<u64>,
 ) -> Result<ChangesetResponse, KintoError> {
     debug!(
-        "The expected timestamp for bucket={}, collection={} is {}",
+        "The expected timestamp for bucket={}, collection={} is {:?}",
         bid, cid, expected
     );
-
     let url = format!(
-        "{}/buckets/{}/collections/{}/changeset?_expected={}",
-        server, bid, cid, expected
+        "{}/buckets/{}/collections/{}/changeset{}",
+        server,
+        bid,
+        cid,
+        match expected {
+            None => String::new(),
+            Some(v) => format!("?_expected={}", v),
+        }
     );
```

Comment:
This is starting to get a bit complicated in my opinion. I'd probably write it more like this, but feel free to keep it as is:

```
let url = {
    let mut partial = format!("{}/buckets/{}/collections/{}/changeset", server, bid, cid);
    if let Some(v) = expected {
        partial.push_str(&format!("?_expected={}", v));
    }
    partial
};
```

Reply:
Great, thanks! This is exactly what I need, idioms :) I'm not used to this kind of block with return...